iothub/device/src/Microsoft.Azure.Devices.Client.csproj

```diff
@@ -52,13 +52,15 @@
   <ItemGroup>
     <PackageReference Include="MQTTnet" Version="4.1.2.350" />
     <PackageReference Include="Microsoft.Azure.Amqp" Version="2.5.12" />
+    <PackageReference Include="System.Text.Json" Version="5.0.0" />
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0-preview1.22518.1">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
     
     <!-- Add support for IAsyncDisposable, match version used in track2 https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L80 -->
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="5.0.0" />
```

Comment:
Adding v5 of STJ requires an update to this too.

Waiting for a response from Azure SDK folks for overall guidance.